src/main/java/com/gtnewhorizons/modularui/api/forge/FluidTankLong.java

```diff
@@ -0,0 +1,230 @@
+package com.gtnewhorizons.modularui.api.forge;
+
+import static com.google.common.primitives.Ints.saturatedCast;
+
+import java.io.IOException;
+
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidRegistry;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidTankInfo;
+import net.minecraftforge.fluids.IFluidTank;
+
+/**
+ * A fluid tank with long capacity and stored amount
+ * 
+ * @author BlueWeabo
+ */
+public class FluidTankLong implements IFluidTank {
+
+    private Fluid fluid;
+    private long storedAmount;
+    private long capacity;
+    private FluidStack internal;
```

Comment:
Problem is for recipes. As we will need to be converting from long to int. There is a chance we will get less than long, unless the recipe map is made to accept FluidTankLong and work with it. Its why the function checks its last stored amount each time. to see if someone drained from it.

Reply:
Ideally we can make `GT_ParallelHelper` use `ItemStackHandler` and `IFluidHandler` to avoid issue like that. But for now `FluidTankLong` is used only for MuTEs (if you address issue with `FluidTankLong(IFluidTank tank)`), so it would be fine.